src/components/reusable_components/CustomSearch.js

```diff
@@ -1,48 +1,19 @@
-import React, { useState, useEffect } from "react";
-import { View, Text, TouchableOpacity, StyleSheet } from "react-native";
+import React, { useState, useEffect, useContext } from "react";
+import { View, Text, StyleSheet, TouchableOpacity } from "react-native";
 import { ActivityIndicator, Searchbar } from "react-native-paper";
 import { THEME } from "../../constants/Theme";
 import { FlashList } from "@shopify/flash-list";
+import AlgoquantApiContext from "../../constants/ApiContext";
 
 export default function CustomSearch(props) {
-  const { searchType, navigation } = props;
-  console.log("Search type", searchType);
-  const mockData = [
-    { name: "Apple", abbreviation: "APPL", price: "$123.90", id: 0 },
-    { name: "Amazon", abbreviation: "AMZN", price: "$112.21", id: 1 },
-    { name: "Google", abbreviation: "GOOGL", price: "$99.21", id: 2 },
-    { name: "Microsoft", abbreviation: "MSFT", price: "$78.21", id: 3 },
-    { name: "Spotify", abbreviation: "SPOT", price: "$90.12", id: 4 },
-  ];
-  const [searchQuery, setSearchQuery] = useState(null);
-  const [searchResults, setSearchResults] = useState(null);
-  const [isLoading, setIsLoading] = useState(false);
-
-  // Filter results whenever we change the query
-  useEffect(() => {
-    queryDatabase();
-  }, [searchQuery]);
-
-  function onChangeSearch(query) {
-    setSearchQuery(query);
-  }
-
-  // Simulates an api call with an artificial loading time
-  function queryDatabase() {
-    setIsLoading(true);
-    console.log("Search query: ", searchQuery);
-    setTimeout(() => {
-      let result = mockData.filter((item) => {
-        if (searchQuery?.length > 0) {
-          return item.name.includes(searchQuery);
-        } else {
-          return null;
-        }
-      });
-      setSearchResults(result);
-      setIsLoading(false);
-    }, 500);
-  }
+  const {
```

Comment:
You should only have to pass in the 

`searchResults` , `getSearchResults` , and a callback for `resettingSearch`

Refer to web on how it is setup.